sphinx_search/static/js/rtd_sphinx_search.js

```diff
@@ -550,28 +554,78 @@ const fetchAndGenerateResults = (api_endpoint, parameters, projectName) => {
+    let innerHTML = `
+        <div class="search__outer">
+          <div class="search__cross" title="Close">
+            <!--?xml version="1.0" encoding="UTF-8"?-->
+            <svg class="search__cross__img" width="15px" height="15px" enable-background="new 0 0 612 612" version="1.1" viewBox="0 0 612 612" xml:space="preserve" xmlns="http://www.w3.org/2000/svg">
+              <polygon points="612 36.004 576.52 0.603 306 270.61 35.478 0.603 0 36.004 270.52 306.01 0 576 35.478 611.4 306 341.41 576.52 611.4 612 576 341.46 306.01"></polygon>
+            </svg>
+          </div>
+          <input class="search__outer__input" placeholder="Search...">
+          <div class="bar"></div>
+          <div class="search__filters">
+            <ul>
+            </ul>
+          </div>
+        </div>
+        <div class="rtd__search__credits">
+          Search by <a href="https://readthedocs.org/">Read the Docs</a> & <a href="https://readthedocs-sphinx-search.readthedocs.io/en/latest/">readthedocs-sphinx-search</a>
+        </div>
+    `;
 
     let div = createDomNode("div", {
         class: "search__outer__wrapper search__backdrop",
     });
     div.innerHTML = innerHTML;
+
+    let filters_list = div.querySelector(".search__filters ul");
+    const config = getConfig();
+    // Add filters below the search box if present.
+    if (filters.length > 0) {
+      let li = createDomNode("li", {"class": "search__filters__title"});
+      li.innerText = "Filters:";
+      filters_list.appendChild(li);
+    }
+    // Each checkbox contains the index of the filter,
+    // so we can get the proper filter when selected.
+    for (let i = 0, len = filters.length; i < len; i++) {
+      const [name, filter] = filters[i];
+      let li = createDomNode("li", {"class": "search__filter", "title": filter});
+      let id = `rtd-search-filter-${i}`;
+      let checkbox = createDomNode("input", {"type": "checkbox", "id": id});
+      let label = createDomNode("label", {"for": id});
+      label.innerText = name;
+      checkbox.value = i;
+      li.appendChild(checkbox);
+      li.appendChild(label);
+      filters_list.appendChild(li);
+
+      checkbox.addEventListener("click", event => {
+        // Uncheck all other filters when one is checked.
+        // We only support one filter at a time.
+        const checkboxes = document.querySelectorAll('.search__filters input');
+        for (const checkbox of checkboxes) {
+          if (checkbox.checked && checkbox.value != event.target.value) {
+            checkbox.checked = false;
+          }
+        }
+
+        // Trigger a search with the current selected filter.
+        let search_query = getSearchTerm();
+        const filter = getCurrentFilter(config);
+        search_query = filter + " " + search_query;
+        const search_params = {
+          q: search_query,
+        };
+        fetchAndGenerateResults(config.api_endpoint, search_params, config.project)();
```

Comment:
Do we want to perform the search immediately when we click on the checkbox? or do we want to press Enter or similar first?

Reply:
Good point 👍🏼